src/ts/core/features/random-page.ts

```diff
@@ -0,0 +1,54 @@
+import {Feature, Settings, Shortcut} from '../settings'
+import {RoamDb} from 'src/core/roam/roam-db'
+import {Roam} from 'src/core/roam/roam'
+
+const DEFAULT_EXCLUDE_PATTERN = ', 2019|, 2020|\\[\\[interval|\\[\\[factor'
+
+const insertRandomLink = async (): Promise<string> => {
+    console.log('INSERT')
+    const excludePattern = await Settings.get('random-page', 'random-page-exclude', DEFAULT_EXCLUDE_PATTERN)
+    const excludeRegex = new RegExp(excludePattern)
+    const pageNames = RoamDb.getAllPages()
+        .filter(({name}) => !name.match(excludeRegex))
+        .map(({name}) => name)
+    const pageName = pageNames[Math.floor(pageNames.length * Math.random())]
```

Comment:
doesn't underscore have function to pick random or something? :)

Reply:
Didn't find a "pick random item", but I did find a "generate random integer"